sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -315,16 +311,11 @@ private[hive] class SparkExecuteStatementOperation(
         } else {
           logError(s"Error executing query with $statementId, currentState $currentState, ", e)
           setState(OperationState.ERROR)
+          HiveThriftServer2.eventManager.onStatementError(
+            statementId, e.getMessage, SparkUtils.exceptionString(e))
           e match {
-            case hiveException: HiveSQLException =>
-              HiveThriftServer2.eventManager.onStatementError(
```

Comment:
now the `onStatementError` is never called?

Reply:
https://github.com/apache/spark/pull/28963/files#diff-424526b50bfb53733a4c2e6c6a3ddd8dR101 it's here

Reply:
and here https://github.com/apache/spark/pull/28963/files/ba0c44ddb558f5b52e23e88f6e6972e5d3192ab4#diff-72dcd8f81a51c8a815159fdf0332acdcR314